frontend/test/specs/comment/commentControllerSpec.js

```diff
@@ -0,0 +1,584 @@
+        });
+    });
+
+    describe('deleteComment', function () {
+        it('should delete the comment', function () {
+            commentCtrl.post.data_comments = [comment];
+            expect(commentCtrl.post.data_comments).toEqual([comment]);
+            spyOn(commentService, 'deleteComment').and.callFake(fakeCallback);
+            spyOn(messageService, 'showToast');
+            commentCtrl.deleteComment();
+            expect(commentService.deleteComment).toHaveBeenCalledWith(post.key, comment.id);
+            expect(messageService.showToast).toHaveBeenCalledWith('Comentário excluído com sucesso');
+            expect(commentCtrl.post.data_comments).toEqual([]);
+        });
+    });
+
+    describe('confirmCommentDeletion()', function(){
+
+        it('Should delete the comment', function() {
+            spyOn(mdDialog, 'confirm').and.callThrough();
+            spyOn(mdDialog, 'show').and.callFake(fakeCallback);
+            spyOn(commentService, 'deleteComment').and.callThrough();
+            commentCtrl.post.data_comments = [comment];
+            httpBackend.expect('DELETE', POSTS_URI + '/' + post.key + '/comments/' + "5").respond(comment);
+            commentCtrl.confirmCommentDeletion("$event");
+            httpBackend.flush();
+            expect(commentService.deleteComment).toHaveBeenCalledWith(commentCtrl.post.key, 5);
+            expect(commentCtrl.post.data_comments).toEqual([]);
+            expect(mdDialog.confirm).toHaveBeenCalled();
+            expect(mdDialog.show).toHaveBeenCalled();
+        });
+
+        it('Should call deleteReply', function() {
+            comment.replies[reply.id] = reply;
+            commentCtrl.reply = reply;
+            commentCtrl.setReplyId();
+            commentCtrl.setCurrentComment();
+            
+            spyOn(mdDialog, 'confirm').and.callThrough();
+            spyOn(mdDialog, 'show').and.callFake(fakeCallback);
+            spyOn(commentService, 'deleteReply').and.callThrough();
+            
+            httpBackend.expect(
+                'DELETE', POSTS_URI + '/' + post.key + '/comments/'+ comment.id +'/replies/'+ reply.id
+            ).respond(reply);
+            commentCtrl.confirmCommentDeletion("$event", reply);
+            httpBackend.flush();
+
+            expect(commentService.deleteReply).toHaveBeenCalledWith(commentCtrl.post.key, comment.id, reply.id);
+            expect(commentCtrl.comment.replies).toEqual({});
+            expect(mdDialog.confirm).toHaveBeenCalled();
+            expect(mdDialog.show).toHaveBeenCalled();
+        });
+    });
+
+    describe('addLike', function() {
+        it('should add a like to the comment', function () {
+            expect(commentCtrl.currentComment.likes).toEqual([]);
+            commentCtrl.addLike();
+            expect(commentCtrl.currentComment.likes).toEqual([user.key]);
+        });
+
+        it('should add a like to the reply', function () {
```

Comment:
since you are only using `currentComment` theres no need for this test. It would be nice to have this test if Comment and Reply have different prototypes.